File: 887_Super_Egg_Drop/887_1.py
from typing import List

class Solution:
    def superEggDrop(self, K: int, N: int) -> int:
        dp = [0 for _ in range(K + 1)]
        m = 0
        while dp[K] < N:
            m += 1
            for k in range(K, 0, -1):
                dp[k] += (dp[k - 1] + 1)
        return m


if __name__ == "__main__":
    s = Solution()
    print(s.superEggDrop(3, 14))

File: 887_Super_Egg_Drop/887_1.c
int superEggDrop(int K, int N){
    int dp[K + 1][N + 1];
    memset(dp, 0, sizeof(dp));
    int m = 0;
    while (dp[K][m] < N) {
        m++;
        for (int k = 1; k <= K; ++k) {
            dp[k][m] = dp[k][m - 1] + dp[k - 1][m - 1] + 1;
        }
    }
    return m;
}

int main(void) {
    printf("%d\n", superEggDrop(3,14));
}